playwright.config.ts

```diff
@@ -1,20 +1,22 @@
 /** @type {import('@playwright/test').PlaywrightTestConfig} */
-import { devices } from '@playwright/test';
+import { devices, PlaywrightTestConfig } from '@playwright/test';
 
-const url = 'https://localhost:3000';
-const config = {
-  webServer: {
+const url = process.env.PLAYWRIGHT_TEST_BASE_URL || 'https://localhost:3000';
+const config: PlaywrightTestConfig = {
+  webServer: process.env.CI ? null : {
     command: 'npm run preview',
     ignoreHTTPSErrors: true,
-    url,
+    reuseExistingServer: true
   },
   use: {
     headless: !!process.env.CI,
     baseURL: `${url}/e2e/`,
     ignoreHTTPSErrors: true,
+    trace: 'on'
   },
+  retries: process.env.CI ? 2 : 0,
```

Comment:
this is what i added for flakiness in ci. have to investigate and maybe we can remove. i think official playwright repo uses this though. 0 means 8 workers in this case or hte max when not on CI.

Reply:
The flakiness is due to not using the localhost server within Github actions?

Reply:
i think its actually more related to the test. the webpage does start up and is waiting. i think its test related.

Reply:
Huh, that's odd. I've actually noticed the tests being _more_ stable locally and on Github actions. So, this seems concerning that we've lost the progress.